app/services/mongodb/MongoDBExtractionService.scala

```diff
@@ -39,6 +41,12 @@ class MongoDBExtractionService extends ExtractionService {
     Extraction.find(MongoDBObject("file_id" -> new ObjectId(resource.id.stringify))).toList
   }
 
+  def findByExtractorIDBefore(extractorID: String, status: String, date: String, limit: Int): List[Extraction] = {
+    val order = MongoDBObject("start"-> -1 )
+    val sinceDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS").parse(date)
+    Extraction.find($and("extractor_id" $eq extractorID, "status" $eq status, "start" $gte sinceDate)).sort(order).limit(limit).toList
+  }
```

Comment:
This still appears to be counting the number of events, not the number of unique jobs. I am not sure how to run [`distinct` queries](https://docs.mongodb.com/manual/reference/method/db.collection.distinct/) with Casbah, but am looking into it now. 

This was pointed out during the review, but never addressed: https://github.com/clowder-framework/clowder/pull/79/files#r499039736

Reply:
You can have access to the underlying casbah collection through the DAO `Extraction.dao.collection.distinct()`.